pkg/scheduler/factory/plugins_test.go

```diff
@@ -136,3 +137,104 @@ func TestBuildScoringFunctionShapeFromRequestedToCapacityRatioArgumentsNilResour
 	assert.Equal(t, expectedShape, builtShape)
 	assert.Equal(t, expectedResources, resources)
 }
+
+func appendToPluginSet(pluginSet *config.PluginSet, name string) *config.PluginSet {
+	if pluginSet == nil {
+		pluginSet = &config.PluginSet{}
+	}
+	pluginSet.Enabled = append(pluginSet.Enabled, config.Plugin{Name: name})
+	return pluginSet
+}
+
+func TestRegisterFrameworkConfigProducers(t *testing.T) {
+	testPredicateName1 := "testPredicate1"
+	testFilterName1 := "testFilter1"
+	RegisterPredicateToFrameworkConfigProducer(testPredicateName1,
```

Comment:
As the producer map are private variables, we can't arbitrarily change their states. Instead, we should use defer() to restore to the initial state in this unit test.

Reply:
I added unregister calls in the defer function.

Reply:
now that they are private to a struct, we don't need to do that.